### PR TITLE
Service as library example finished

### PR DESCRIPTION
**Pull request by juanlu-rti**
Wrapping up the Service as library example.

### CODE REVIEW

**Comment by juanlu-rti** [comment]
@titosanxez , @iblancasa (CMake), could you review the changes?

File: examples/recording_service/service_as_lib/cpp/ServiceAsLibExample.cxx [hunk 1]
```diff
@@ -11,24 +11,96 @@
...
+    const std::string recorder_role("record");
+    const std::string replay_role("replay");
 
+    // Check if a role argument has been provided
+    if (argc > 1) {
+        if (recorder_role.compare(argv[1]) == 0) {
+            service_property.application_role(
+                    rti::recording::ApplicationRoleKind::RECORD_APPLICATION);
+            service_property.cfg_file("recorder_config.xml");
+        } else if (recorder_role.compare(argv[1]) == 0) {
```
Comment (titosanxez) [inline examples/recording_service/service_as_lib/cpp/ServiceAsLibExample.cxx]: do you mean replay_role?

File: examples/recording_service/service_as_lib/cpp/ServiceAsLibExample.cxx [hunk 2]
```diff
@@ -11,24 +11,96 @@
...
+    try {
+        process_role(argc, argv, service_property, running_seconds);
+    } catch (const std::exception& ex) {
+        std::cerr << "Exception: " << ex.what() << std::endl;
+        print_usage(argv[0]);
+        return EXIT_FAILURE;
+    }
+    // The service configuration, regardless of the role, is always the same
+    service_property.service_name("service_as_lib");
+    service_property.enable_administration(true);
```
Comment (titosanxez) [inline examples/recording_service/service_as_lib/cpp/ServiceAsLibExample.cxx]: I would enable these only if their ids are provided

File: examples/recording_service/service_as_lib/cpp/recorder_config.xml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+            <statistics_sampling_period>
+                <sec>0</sec>
+                <nanosec>500000000</nanosec>
+            </statistics_sampling_period>
+        </monitoring>
+
+        <!--  Top-level storage settings -->
+        <storage>
+            <sqlite>
+                <storage_format>XCDR_AUTO</storage_format>
```
Comment (titosanxez) [inline examples/recording_service/service_as_lib/cpp/recorder_config.xml]: lets simplify defaults

File: examples/recording_service/service_as_lib/cpp/ServiceAsLibExample.cxx (hunk 1 above)
Comment (juanlu-rti) [reply examples/recording_service/service_as_lib/cpp/ServiceAsLibExample.cxx]: Yes, it was corrected yesterday.

File: examples/recording_service/service_admin/cpp/CMakeLists.txt [hunk 4]
```diff
@@ -10,7 +10,8 @@
 #  use or inability to use the software.
 #
 cmake_minimum_required(VERSION 3.11)
-project (RecorderRemoteTimeTagging)
+set (CMAKE_CXX_STANDARD 11)
```
Comment (iblancasa) [inline examples/recording_service/service_admin/cpp/CMakeLists.txt]: Apply this as a target property. https://cmake.org/cmake/help/latest/command/set_target_properties.html

File: examples/recording_service/service_as_lib/cpp/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+#  RTI grants Licensee a license to use, modify, compile, and create derivative
+#  works of the Software.  Licensee has the right to distribute object form
+#  only for use with RTI products.  The Software is provided "as is", with no
+#  warranty of any type, including any warranty for fitness for any purpose.
+#  RTI is under no obligation to maintain or support the Software.  RTI shall
+#  not be liable for any incidental or consequential damages arising out of the
+#  use or inability to use the software.
+#
+cmake_minimum_required(VERSION 3.11)
+set (CMAKE_CXX_STANDARD 11)
```
Comment (iblancasa) [inline examples/recording_service/service_as_lib/cpp/CMakeLists.txt]: Same case as in the other CMakeList: set this as a target property

File: examples/recording_service/service_as_lib/cpp/CMakeLists.txt [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+
+find_package(
+    RTICodeGenerator "3.0.0"
+    REQUIRED
+)
+
+# Select the adequate version of the Recording Service library depending on the
+# build type
+set(recording_lib "rtirecordingservice")
+if(NOT CMAKE_BUILD_TYPE)
```
Comment (iblancasa) [inline examples/recording_service/service_as_lib/cpp/CMakeLists.txt]: Could you move this to the top? It is weird to have this ``if`` statement here when you are setting the correct name for the library

File: examples/recording_service/service_as_lib/cpp/CMakeLists.txt [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+    ServiceAsLibExample
+    "${CMAKE_CURRENT_SOURCE_DIR}/ServiceAsLibExample.cxx"
+)
+
+add_dependencies(ServiceAsLibExample copy_config_files)
+
+target_include_directories(
+    ServiceAsLibExample
+    PUBLIC 
+        "${CMAKE_BINARY_DIR}"
```
Comment (iblancasa) [inline examples/recording_service/service_as_lib/cpp/CMakeLists.txt]: ``CMAKE_CURRENT_BINARY_DIR``.

Also, why is this needed? I don't see what are you including from the binary directory.

File: examples/recording_service/service_as_lib/cpp/ServiceAsLibExample.cxx (hunk 2 above)
Comment (titosanxez) [reply examples/recording_service/service_as_lib/cpp/ServiceAsLibExample.cxx]: @juanlu-rti,  this is the only change left I haven't seen in the final PR
